src/components/calcite-inline-editable/calcite-inline-editable.e2e.ts

```diff
@@ -0,0 +1,274 @@
+      await input.type("Moe");
+      await confirmChangesButton.click();
+      expect(calciteInlineEditableConfirmChanges).toHaveReceivedEventTimes(1);
+      await page.waitForChanges();
+      expect(input.getAttribute("value")).toBe("John DoeMoe");
+      expect(element).not.toHaveAttribute("editing-enabled");
+    });
+
+    it("it does not disable editing when onConfirmChanges resolves unsuccessfully", async () => {
+      const element = await page.find("calcite-inline-editable");
+      const onConfirmChanges = () => new Promise((_resolve, reject) => setTimeout(reject, 100));
+      // https://github.com/ionic-team/stencil/issues/1174
+      page.exposeFunction("onConfirmChanges", onConfirmChanges);
+      page.$eval("calcite-inline-editable", (el: HTMLCalciteInlineEditableElement) => {
+        el.onConfirmChanges = onConfirmChanges;
+      });
+      const calciteInlineEditableConfirmChanges = await page.spyOnEvent("calciteInlineEditableConfirmChanges");
+      const input = await page.find("calcite-input");
+      const enableEditingButton = await element.find(".calcite-inline-editable-enable-editing-button");
+      await enableEditingButton.click();
+      const confirmChangesButton = await element.find(".calcite-inline-editable-confirm-changes-button");
+      await input.type("Moe");
+      await confirmChangesButton.click();
+      expect(calciteInlineEditableConfirmChanges).toHaveReceivedEventTimes(1);
+      await page.waitForChanges();
+      expect(input.getAttribute("value")).toBe("John DoeMoe");
+      expect(element).toHaveAttribute("editing-enabled");
+    });
+
+    describe("with label", () => {
+      beforeEach(async () => {
+        page = await newE2EPage();
+        await page.setContent(`
+        <calcite-label>
+          <span>Hello</span>
+          <calcite-inline-editable has-controls>
+            <calcite-input value="John Doe"/>
+          </calcite-inline-editable>
+        </calcite-label>
+        `);
+      });
+
+      it("focuses the enable editing button when the label is clicked", async () => {
+        const label = await page.find("span");
+        await label.click();
+        const activeEl = await page.evaluateHandle(() => document.activeElement);
+        expect(activeEl["_remoteObject"].description).toMatch(
+          "calcite-button.calcite-inline-editable-enable-editing-button"
+        );
+      });
+
+      it("focuses the input when editing is enabled and the label is subsequently clicked", async () => {
+        const enableEditingButton = await page.find(".calcite-inline-editable-enable-editing-button");
+        await enableEditingButton.click();
+        const label = await page.find("span");
+        await label.click();
+        const activeEl = await page.evaluateHandle(() => document.activeElement);
+        expect(activeEl["_remoteObject"].description).toMatch("input");
+      });
+    });
+  });
+});
```

Comment:
☝️ missing an `accessible()` test.  You can reference other e2e tests to see them in use.

Reply:
Good catch. Didn't know about that handy helper. Pretty nifty.